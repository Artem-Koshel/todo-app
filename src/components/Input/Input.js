import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import styles from './Input.module.scss';

export default function Input({ onEnterPress }) {
  const [value, setValue] = useState('');

  const handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      onEnterPress(value);
      setValue('');
    }
  };
  return (
    <input
      className={styles.todoInput}
      onKeyUp={handleKeyPress}
      type="text"
      value={value}
      onChange={(event) => setValue(event.target.value)}
      placeholder="Input your text and press ENTER"
    />
  );
}

Input.propTypes = {
  onEnterPress: PropTypes.func.isRequired,
};
