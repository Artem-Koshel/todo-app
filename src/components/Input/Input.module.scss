.todoInput {
    height: 50px;
    border-radius: 5px;
    box-shadow: 1px 1px 50px lightgray;
    border: 0;
    padding: 2px 15px;
    font-family: inherit;
    font-size: large;
    outline: none;

    &::placeholder {
        opacity: .4;
        font-size: large;
        font-family: inherit;
    }
}