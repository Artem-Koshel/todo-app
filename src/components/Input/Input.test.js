import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

describe('Input component', () => {
  it('render correctly', () => {
    const { container } = render(<Input onEnterPress={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });

  it('exec callback on Enter press', async () => {
    const user = userEvent.setup();
    const callback = jest.fn();
    render(<Input onEnterPress={callback} />);

    await user.click(screen.getByRole('textbox'));
    await user.keyboard('test string{Enter}');

    expect(callback).toBeCalledWith('test string');
  });

  it('clear value after Enter press', async () => {
    const user = userEvent.setup();
    render(<Input onEnterPress={jest.fn()} />);

    await user.click(screen.getByRole('textbox'));
    await user.keyboard('test string{Enter}');

    const input = await screen.findByRole('textbox');
    expect(input.value).toBe('');
  });
});
