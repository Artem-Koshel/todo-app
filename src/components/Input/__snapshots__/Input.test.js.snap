// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render correctly 1`] = `
<div>
  <input
    class="todoInput"
    placeholder="Input your text and press ENTER"
    type="text"
    value=""
  />
</div>
`;
